backend/case/src/main/java/org/pucar/dristi/util/CaseUtil.java

```diff
@@ -34,4 +48,34 @@ public static String generateAccessCode(int length) {
     public Long getCurrentTimeMil() {
         return System.currentTimeMillis();
     }
+
+    public String generateCNRNumber(String tenantId, String userID) {
+        Map<String,Object> resultMap = caseRepository.searchSeqNum("pg", COURTID);
+        String newSqNumb = "";
+
+        if(resultMap!=null) {
+            String seqNum = resultMap.get("cnr_seq_num").toString();
+            String createdTime = resultMap.get("created_time").toString();
+
+            long currentTimeMillis = System.currentTimeMillis();
+            LocalDateTime currentDateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(currentTimeMillis), ZoneId.of(ZONE_ID));
+
+            LocalDateTime lastInsertedDateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(Long.parseLong(createdTime)), ZoneId.of(ZONE_ID));
+
+            int currentYear = currentDateTime.getYear();
+            int lastInsertedYear = lastInsertedDateTime.getYear();
+
+            if (currentYear != lastInsertedYear) {
+                newSqNumb = "000001";
+                caseRepository.insertSeqNum(UUID.randomUUID(), tenantId, COURTID, newSqNumb,userID);
+            } else {
```

Comment:
Please explain with comment on what we are trying to do here. Note that DB related updates need to happen in a transaction or we should use the PostgreSQL's SEQUENCE Number concept. but the reset logic to restart from 1 should happen in a cron job and not from code, because multiple parties may call the code at the same time.